wire-ios/Wire-iOS/Sources/UserInterface/UserProfile/ProfileViewController.swift

```diff
@@ -624,4 +616,8 @@ extension ProfileViewController: ProfileViewControllerViewModelDelegate {
     func stopAnimatingActivity() {
         activityIndicator.stopAnimating()
     }
+
+    func transition(to conversation: ZMConversation) {
+        delegate?.profileViewController(self, wantsToNavigateTo: conversation)
```

Comment:
I think this `ProfileViewControllerDelegate` is needed to manage navigation, isn't it?

From my understanding this might be better to have the logic in view model and call it from here:
`viewModel.navigateTo(conversation)`.

The view model can have a closure `var onNavigation(to conversation: Conversation)`, that is set from the outer world, or another delegate for navigation only.

Reply:
Agreed. I originally just wanted the view model to delegate the navigation _somewhere_ in order to mock it (the delegate) for tests. 

Where would you set the closure from? I guess from where we have an instance of the VM, but that would be the VC. Wouldn't it be the same as having that navigation logic in the VC with extra steps?

Or, having another delegate could work, I was thinking of something like a router. WDYT?

Reply:
without more structure, I would use convenience init to setup the viewModel and the closure for navigation.

Otherwise, move the logic to a separate object like a Builder

Reply:
```
protocol ProfileViewControllerViewModeling {
// ...
/// set from the outer world to communicate
var onNavigationToConversation: ((ZMConversation) -> Void)? { get set }
/// called from the view to forward to the closure
func navigate(to conversation: ZMConversation)
```

But I see the problem now, the caller side doesn't create a `viewModel` and runs `ProfileViewController(viewModel: viewModel)`, but we have a lot of convenience init and stuff around, that is not easy to refactor. 
An example is `UserDetailViewControllerFactory.createUserDetailViewController`.

Not sure if we just keep it as it is (one can understand it well) or pass a closure through all the inits instead of the delegate...

Reply:
https://github.com/wireapp/wire-ios/pull/1365/commits/c8e57081d54b03ce21f7cd4b9c318faab9519978

Reply:
went with the closure solution